src/platform/SDL2Window.cxx

```diff
@@ -259,7 +283,32 @@ bool SDLWindow::create(void)
             return false;
         }
     }
-#endif
+
+    // Depending on the distribution (or possibly the window manager), SDL will not recognize the new window resolution
+    // and will keep returning the previous resolution when SDL_GetWindowSize() is called, until a period of time has
+    // passed and events have been pumped. Wait up to two seconds for the correct resolution to start being returned,
+    // checking every quarter second, to avoid repeating the window destruction/re-creation process based on bad data.
```

Comment:
I wonder if there is a way to flush the queue? This seems like an exceptionally awkward work-around. Or maybe we don't understand the resize event logic?

Reply:
Completely agreed that the workaround is awkward. `SDL_PumpEvents()` *is* flushing the queue in this case. After extensive testing, it appeared to me that *both* a time delay *and* pumping events are required in order to reach the point where the correct window resolution is returned. My best guess is that the window creation process in SDL returns before the modeset is complete. It took extensive testing to get to this point. However, if anyone else has a better solution, I'd be all for it.